src/app/devices/deviceList/components/deviceList.spec.tsx

```diff
@@ -5,48 +5,81 @@
 import 'jest';
 import { shallow } from 'enzyme';
 import * as React from 'react';
-import DeviceListComponent from './deviceList';
-import { testWithLocalizationContext } from '../../../shared/utils/testHelpers';
+import { CommandBar } from 'office-ui-fabric-react/lib/CommandBar';
+import DeviceListComponent, { DeviceListDataProps, DeviceListDispatchProps } from './deviceList';
+import { testSnapshot, mountWithLocalization } from '../../../shared/utils/testHelpers';
 import { DeviceSummary } from '../../../api/models/deviceSummary';
 
+const pathname = `/`;
+
+const location: any = { // tslint:disable-line:no-any
+    pathname
+};
+const routerprops: any = { // tslint:disable-line:no-any
+    history: {
+        location
+    },
+    location,
+    match: {}
+};
+
 const devices: DeviceSummary[] = [
     {
         authenticationType: 'sas',
         cloudToDeviceMessageCount: '0',
+        connectionState: 'connected',
         deviceId: 'testDeviceId',
+        iotEdge: false,
         lastActivityTime: '0001-01-01T00:00:00Z',
         status: 'Enabled',
         statusUpdatedTime: '0001-01-01T00:00:00Z'
     }
 ];
-jest.mock('./deviceListCell', () => <></>);
+
+const deviceListDataProps: DeviceListDataProps = {
+    connectionString: 'connectionString',
+    devices,
+    isFetching: false
+};
+
+const mockListDevices = jest.fn();
+const deviceListDispatchProps: DeviceListDispatchProps = {
+    deleteDevices: jest.fn(),
+    listDevices: mockListDevices
+};
+
+const getComponent = (overrides = {}) => {
+    const props = {
+        ...deviceListDataProps,
+        ...deviceListDispatchProps,
+        ...routerprops,
+        ...overrides
+    };
+
+    return (
+        <DeviceListComponent {...props} />
+    );
+};
 
 describe('components/devices/DeviceList', () => {
     it('matches snapshot', () => {
-        const wrapper = testWithLocalizationContext(
-        <DeviceListComponent
-            connectionString="connectionString"
-            devices={devices}
-            listDevices={jest.fn()}
-            deleteDevices={jest.fn()}
-            isFetching={false}
-        />);
-
-        expect(wrapper).toMatchSnapshot();
+        testSnapshot(getComponent());
```

Comment:
Would be good to test snapshots with `query.continuationTokens` and `query.currentPageIndex`

Reply:
Also with rendering `lastActivityTime` and `statusUpdatedTime` rendering as `'--'`